src/error.rs

```diff
@@ -252,24 +252,53 @@ impl StdError for ParseError {
     }
 }
 
-/// An [invalid character](https://tools.ietf.org/html/rfc3501#section-4.3) was found in an input
-/// string.
+/// An [invalid character](https://tools.ietf.org/html/rfc3501#section-4.3) was found in a command
+/// argument.
 #[derive(Debug)]
-pub struct ValidateError(pub char);
+pub struct ValidateError {
+    /// the synopsis of the invalid command
+    pub command_synopsis: String,
+    /// the name of the invalid argument
+    pub argument: String,
+    /// the invalid character contained in the argument
+    pub offending_char: char,
+}
```

Comment:
I think we should make these all non-`pub` — they're not really useful to users except insofar as they are included in the printed representation.

This is also a breaking change because previously this struct had only public fields, and thus had a public constructor and could be matched against, which is kind of awkward. I don't know if anyone previously actually _inspected_ a `ValidateError`, but it's possible. I think I'm okay with including this in a minor release though since we're still in alpha.

Reply:
Ok, the fields are `pub(crate)` now.

Thanks for the hint about the breaking change... I totally missed that.